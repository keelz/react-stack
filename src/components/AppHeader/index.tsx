import * as React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../common/assets/img/logo.svg';

const AppHeader: React.SFC = () =>
  <header className="App-header">
    <div className="App-navigator">
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/about">About</Link></li>
        <li><Link to="/topics">Topics</Link></li>
      </ul>
    </div>
    <img src={logo} className="App-logo" alt="logo" />
    <h1 className="App-title">Welcome to React</h1>
  </header>;

export default AppHeader;
