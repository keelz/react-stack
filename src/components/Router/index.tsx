import * as React from 'react';
import About from '../About';
import Home from '../Home';
import Topics from '../Topics';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';

const AppRouter: React.SFC = props =>
  <div className="Router">
    <Router>
      <div>
        { props.children }
        <Route exact path="/" component={Home} />
        <Route path="/about" component={About} />
        <Route path="/topics" component={Topics} />
      </div>
    </Router>
  </div>;

export default AppRouter;
