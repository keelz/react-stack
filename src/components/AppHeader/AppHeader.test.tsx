import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Router from '../Router';
import AppHeader from './index';

const adapter = new Adapter();
configure({ adapter });

describe('AppHeader suite', () => {
  it('should match snapshot', () => {
    const tree = shallow(<AppHeader />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should be selectable by class .App-header', () => {
    expect(shallow(<AppHeader />).is('.App-header')).toBe(true);
  });

  it('should mount within a Router component', () => {
    expect(mount(<Router><AppHeader /></Router>).find('.App-header').length).toBe(1);
  });
});
