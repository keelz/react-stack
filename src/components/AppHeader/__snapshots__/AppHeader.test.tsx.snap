// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader suite should match snapshot 1`] = `
<header
  className="App-header"
>
  <div
    className="App-navigator"
  >
    <ul>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/about"
        >
          About
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/topics"
        >
          Topics
        </Link>
      </li>
    </ul>
  </div>
  <img
    alt="logo"
    className="App-logo"
  />
  <h1
    className="App-title"
  >
    Welcome to React
  </h1>
</header>
`;
